mqtt/mqtt-bridge/src/persist/memory/loader.rs

```diff
@@ -0,0 +1,431 @@
+use std::{pin::Pin, sync::Arc, task::Context, task::Poll, vec::IntoIter};
+
+use futures_util::stream::Stream;
+use mqtt3::proto::Publication;
+use parking_lot::{Mutex, MutexGuard};
```

Comment:
q: what's the benefit of `parking_lot`? if they are better than std, should we use them everywhere? /cc @dmolokanov

Reply:
Parking lot has fairness enforcement. Denis recommended it over the mutex from std